### PR TITLE
various improvements/corrections to the multisegment well model

### PR DESCRIPTION
**Pull request by GitPaean**
And also trying to activate the multisegment well model by default. 

With multisegment well model, more parameters can be tuned and tested, there might be some small updates base on the testing result. 

Creating this PR to warp up the efforts to improve the multisegment well model for now.

### CODE REVIEW

**Comment by GitPaean** [comment]
With multisegment well model, the relaxation of tolerance (see https://github.com/OPM/opm-simulators/pull/1584) impacts the performance in a significant way.  We are at the point to make a decision about the tolerance or we try to choose different criteria for the convergence checking. 

At the moment, the required tolerance is affected by the compressibility of the field.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by alfbr** [comment]
benchmark

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=286

**Comment by GitPaean** [comment]
benchmark

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=288

**Comment by alfbr** [comment]
Seems there was a hick-up, let's try again:
benchmark

**Comment by alfbr** [comment]
and one more time:
benchmark

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=289

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=290

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                              | Relative |
| ------- | ------------------------------------------ | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1 | 0.965    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8 | 0.954    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1 | 0.996    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8 | 1.034    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=292

**Comment by alfbr** [comment]
Please rebase, we need to build it again for testing.

**Comment by GitPaean** [comment]
done rebasing!

jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this update_data please

File: opm/autodiff/BlackoilWellModel.hpp [hunk 1]
```diff
@@ -358,7 +358,7 @@ namespace Opm {
             void computeRepRadiusPerfLength(const Grid& grid, Opm::DeferredLogger& deferred_logger);
 
 
-            void computeAverageFormationFactor(std::vector<double>& B_avg) const;
+            void computeAverageFormationFactor(std::vector<Scalar>& B_avg) const;
```
Comment (GitPaean) [inline opm/autodiff/BlackoilWellModel.hpp]: I do not recall which function made me change the `double` to `Scalar` here. Hopefully it is not a bid deal.

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
I am done updating.  This is ready to go in now. 

The rest part of the development will be proposed as a new PR.

**Comment by akva2** [comment]
please merge so i can get the packages out the door.

**Comment by GitPaean** [comment]
This PR has been tested with ensemble, it significantly improves the capacity of the MSW. There are some more development coming as a separate PR.  Merging it now for the release.

**Pull request merged**